google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -343,6 +343,16 @@ public JobId get() {
     return create(jobInfo, idProvider, options);
   }
 
+  @Override
+  public Connection createConnection(ConnectionSettings connectionSettings)
+      throws BigQueryException {
+    switch (connectionSettings.getResponseRowFormat()) {
+      default:
+        // TODO: DETERMINE DEFAULT ROW SERIALIZATION FORMAT
```

Comment:
legacy format: we control, but possibly slower.
avro, arrow: potentially faster, but then you've coupled this API closely to the details of the chosen format and have to have affordances for how the formats evolve.

Also, I'd be hesistant about arrow for another reason: the nature of the data representation is columnar, and most classic DB consumers are more familiar with a row-wise format.

Reply:
User could potentially also pass in a converter which converts from one type to another.